client/my-sites/sidebar-unified/index.jsx

```diff
@@ -0,0 +1,74 @@
+/**
+ * MySitesSidebarUnified
+ *   Renders the Sidebar for "My Sites", except all of the menus and items are
+ *   driven off a WPCom endpoint: /sites/${sideId}/admin-menu, which is loaded
+ *   into state.adminMenu in a data layer.
+ *
+ *    Currently experimental/WIP.
+ **/
+
+/**
+ * External dependencies
+ */
+import React, { useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import { getCurrentRoute } from 'state/selectors/get-current-route';
+import { getSelectedSite } from 'state/ui/selectors';
+import { isUnderDomainManagementAll } from 'my-sites/domains/paths';
+import { isUnderEmailManagementAll } from 'my-sites/email/paths';
+import { requestAdminMenu } from '../../state/admin-menu/actions';
+import CurrentSite from 'my-sites/current-site';
+import MySitesSidebarUnifiedItem from './item';
+import MySitesSidebarUnifiedMenu from './menu';
+import Sidebar from 'layout/sidebar';
+
+export const MySitesSidebarUnified = ( { path } ) => {
+	const reduxDispatch = useDispatch();
+	const selectedSite = useSelector( ( state ) => getSelectedSite( state ) );
+	useEffect( () => {
+		if ( selectedSite !== null && selectedSite.ID ) {
+			reduxDispatch( requestAdminMenu( selectedSite.ID ) );
+		}
+	}, [ reduxDispatch, selectedSite ] );
+
+	// Extract this?
+	const menuItems = useSelector( ( state ) => {
+		const thisSelectedSite = getSelectedSite( state );
+		if (
+			thisSelectedSite !== null &&
+			thisSelectedSite.ID &&
+			state.adminMenu != null &&
+			thisSelectedSite.ID in state.adminMenu
+		) {
+			return Object.values( state.adminMenu[ thisSelectedSite.ID ] );
```

Comment:
We will be able to use selectors from https://github.com/Automattic/wp-calypso/pull/45576 soon.

Reply:
@mreishus If you rebase you can now update to use the selectors.

Reply:
Done, there's still some 'glue' though